bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/DateTimeType.java

```diff
@@ -89,18 +92,23 @@ public int hashCode() {
 
     @Override
     public boolean equals(Object obj) {
-        if (this == obj)
+        if (this == obj) {
             return true;
-        if (obj == null)
+        }
+        if (obj == null) {
             return false;
-        if (getClass() != obj.getClass())
+        }
+        if (getClass() != obj.getClass()) {
             return false;
+        }
         DateTimeType other = (DateTimeType) obj;
         if (calendar == null) {
-            if (other.calendar != null)
+            if (other.calendar != null) {
                 return false;
-        } else if (!calendar.equals(other.calendar))
+            }
+        } else if (!calendar.equals(other.calendar)) {

```

Comment:
`Calendar.equals()` requires an exact match, per the doc: https://docs.oracle.com/javase/7/docs/api/java/util/Calendar.html#equals(java.lang.Object)

> The result is true if and only if the argument is a Calendar object of the same calendar system that represents the same time value (millisecond offset from the Epoch) under the same Calendar parameters as this object.
The Calendar parameters are the values represented by the isLenient, getFirstDayOfWeek, getMinimalDaysInFirstWeek and getTimeZone methods. If there is any difference in those parameters between the two Calendars, this method returns false.

> Use the compareTo method to compare only the time values.

This is a problem because `DateTimeType` is not immutable, since `getCalender()` returns the mutable `calendar` member instead of a clone.